pyartifactory/objects.py

```diff
@@ -586,78 +577,70 @@ def deploy(
         """
         artifact_path = artifact_path.lstrip("/")
         local_filename = artifact_path.split("/")[-1]
-        try:
-            with open(local_file_location, "rb") as f:
-                form = encoder.MultipartEncoder(
-                    {
-                        "documents": (local_filename, f, "application/octet-stream"),
-                        "composite": "NONE",
-                    }
-                )
-                headers = {"Prefer": "respond-async", "Content-Type": form.content_type}
-                self._put(f"{artifact_path}", headers=headers, data=form)
-                logging.info(f"Artifact {local_filename} successfully deployed")
-                return self.properties(artifact_path)
-        except ArtifactDeployException:
-            logging.error(f"Cannot deploy artifact {local_filename}")
-            raise
+        with open(local_file_location, "rb") as f:
+            form = encoder.MultipartEncoder(
+                {
+                    "documents": (local_filename, f, "application/octet-stream"),
+                    "composite": "NONE",
+                }
+            )
+            headers = {"Prefer": "respond-async", "Content-Type": form.content_type}
+            self._put(f"{artifact_path}", headers=headers, data=form)
+            logging.info(f"Artifact {local_filename} successfully deployed")
+            return self.properties(artifact_path)
 
-    def download(self, artifact_path: str) -> str:
+    def download(self, artifact_path: str, local_file_path: str = None) -> str:
         """
         :param artifact_path: Path to file in Artifactory
+        :param local_file_path: Local path to where the artifact will be imported
         :return: File name
         """
         artifact_path = artifact_path.lstrip("/")
         local_filename = artifact_path.split("/")[-1]
-        try:
-            with self._get(f"{artifact_path}", stream=True) as r:
-                with open(local_filename, "wb") as f:
-                    for chunk in r.iter_content(chunk_size=8192):
-                        if chunk:  # filter out keep-alive new chunks
-                            f.write(chunk)
-                            # f.flush()
-            logging.info(f"Artifact {local_filename} successfully downloaded")
-            return local_filename
-        except ArtifactDownloadException:
-            logging.error(f"Cannot download artifact {local_filename}")
-            raise
+
+        if local_file_path:
+            Path(local_file_path).mkdir(parents=True, exist_ok=True)
+            local_file_full_path = f"{local_file_path}/{local_filename}"

```

Comment:
If you call your method parameter local_*file*_path, I expect the artifact to be saved under this exact name, and not to create a directory called `local_file_path` in which the artifact will be saved.